apps/nsqd/options.go

```diff
@@ -73,14 +63,45 @@ func (t *tlsMinVersionOption) String() string {
 	return strconv.FormatInt(int64(*t), 10)
 }
 
+type config map[string]interface{}
+
+// Validate settings in the config file, and fatal on errors
+func (cfg config) Validate() {
+	// special validation/translation
+	if v, exists := cfg["tls_required"]; exists {
+		var t tlsRequiredOption
+		err := t.Set(fmt.Sprintf("%v", v))
+		if err == nil {
+			cfg["tls_required"] = t.String()
+		} else {
+			logFatal("failed parsing tls_required %+v", v)
+		}
+	}
+	if v, exists := cfg["tls_min_version"]; exists {
+		var t tlsMinVersionOption
+		err := t.Set(fmt.Sprintf("%v", v))
+		if err == nil {
+			newVal := fmt.Sprintf("%v", t.Get())
+			if newVal != "0" {
+				cfg["tls_min_version"] = newVal
+			} else {
+				delete(cfg, "tls_min_version")
+			}
+		} else {
+			logFatal("failed parsing tls_min_version %+v", v)
+		}
+	}
+}
+
 func nsqdFlagSet(opts *nsqd.Options) *flag.FlagSet {
 	flagSet := flag.NewFlagSet("nsqd", flag.ExitOnError)
 
 	// basic options
 	flagSet.Bool("version", false, "print version string")
 	flagSet.String("config", "", "path to config file")
 
-	flagSet.String("log-level", "info", "set log verbosity: debug, info, warn, error, or fatal")
+	logLevel := lg.LogLevelOption(opts.LogLevel)
+	flagSet.Var(&logLevel, "log-level", "set log verbosity: debug, info, warn, error, or fatal")
 	flagSet.String("log-prefix", "[nsqd] ", "log message prefix")
 	flagSet.Bool("verbose", false, "deprecated in favor of log-level")
```

Comment:
might want to clarify "verbose" is now "ignored - use log-level"
(I dunno if the `Verbose` options struct member can be removed?)

(moved this comment from the commit to the PR)

Reply:
yep